assists/baremetalconfig_xlnx.py

```diff
@@ -0,0 +1,292 @@
+    addr = Lopper.property_get(fdt, intr_parent, 'reg', LopperFmt.COMPOUND)
+    reg,size = scan_reg_size(fdt, intr_parent, addr, 0)
+    return reg
+
+class DtbtoCStruct(object):
+    def __init__(self, out_file):
+        self._outfile = open(out_file, 'w')
+        self._lines = []
+
+    def out(self, line):
+        """Output a string to the output file
+
+        Args:
+            line: String to output
+        """
+        self._outfile.write(line)
+
+    def buf(self, line):
+        """Buffer up a string to send later
+
+        Args:
+            line: String to add to our 'buffer' list
+        """
+        self._lines.append(line)
+
+    def get_buf(self):
+        """Get the contents of the output buffer, and clear it
+
+        Returns:
+            The output buffer, which is then cleared for future use
+        """
+        lines = self._lines
+        self._lines = []
+        return lines
+
+def is_compat(node, compat_string_to_test):
+    if re.search( "baremetaldrv,config", compat_string_to_test):
+        return xlnx_generate_bm_config
+    return ""
+
+# tgt_node: is the baremetal config top level domain node number
+# sdt: is the system device-tree
+# options: baremetal driver meta-data file path
+def xlnx_generate_bm_config(tgt_node, sdt, options):
+    root_node = sdt.tree[tgt_node]
+    root_sub_nodes = root_node.subnodes()
+    node_list = []
+    # Traverse the tree and find the nodes having status=ok property
+    for node in root_sub_nodes:
+        try:
+            status = node["status"].value
+            if "okay" in status:
+                node_list.append(node)
+        except:
+           pass
+
+    sdt.FDT = Lopper.dt_to_fdt(sdt.dtb, 'rb')
+
+    yaml_file = options['args']
+    yamlfile = yaml_file[0]
+    driver_compatlist = []
+    driver_proplist = []
+    # Read the yaml file and get the driver supported compatible list
+    # and config data file required driver properties
+    with open(yamlfile, 'r') as stream:
+        schema = yaml.safe_load(stream)
+        driver_compatlist = compat_list(schema)
+        driver_proplist = schema['required']
+
+    driver_nodes = []
+    for compat in driver_compatlist:
+        for node in node_list:
+           compat_string = node['compatible'].value[0]
+           if compat in compat_string:
+               driver_nodes.append(node) 
+
+    # config file name: x<driver_name>_g.c 
+    driver_name = yamlfile.split('/')[-1].split('.')[0]
+    config_struct = str("X") + driver_name.capitalize() + str("_Config")
+    outfile = str("x") + driver_name + str("_g.c")
+
+    plat = DtbtoCStruct(outfile)
+    for index,node in enumerate(driver_nodes):
+        if index == 0:
+            plat.buf('#include "x%s.h"\n' % driver_name)
+            plat.buf('\n%s %s __attribute__ ((section (".drvcfg_sec"))) = {\n' % (config_struct, config_struct + str("Table[]")))
+        prop_dict = Lopper.node_properties_as_dict(sdt.FDT, node.abs_path)
```

Comment:
I had left the dictionary routines in the Lopper class as a transition. You can always have dictionary access to the properties using the tree nodes that are passed in. Fundamentally, we don't want fdt or other low level access in the assists, you can use the LopperTree methods exclusively.  If something is missing, it can be added.

Reply:
Ok will update the code as you suggested and let you know if i found anything missing.